msal/wam.py

```diff
@@ -0,0 +1,189 @@
+            # https://docs.microsoft.com/en-us/azure/active-directory/devices/troubleshoot-device-dsregcmd
+        "error_description": "{}. Status: {}, Error code: {}, Tag: {}".format(
+            context,
+            error.get_status(), error.get_error_code(), error.get_tag()),
+        }
+
+
+def _read_account_by_id(account_id):
+    """Return the callback result which contains the account or error"""
+    callback_data = _CallbackData()
+    pymsalruntime.read_account_by_id(
+        account_id,
+        "correlation_id",
+        lambda result, callback_data=callback_data: callback_data.complete(result)
+        )
+    callback_data.signal.wait()
+    return callback_data.auth_result
+
+
+def _convert_result(result, client_id):  # Mimic an on-the-wire response from AAD
+    error = result.get_error()
+    if error:
+        return _convert_error(error, client_id)
+    id_token_claims = json.loads(result.get_id_token()) if result.get_id_token() else {}
+    account = result.get_account()
+    assert account, "Account is expected to be always available"
+    ## Note: As of pymsalruntime 0.1.0, only wam_account_ids property is available
+    #account.get_account_property("wam_account_ids")
+    return_value = {k: v for k, v in {
+        "access_token": result.get_access_token(),
+        "expires_in": result.get_access_token_expiry_time(),
+        "id_token_claims": id_token_claims,
+        "client_info": account.get_client_info(),
+        "_account_id": account.get_account_id(),
+        }.items() if v}
+    granted_scopes = result.get_granted_scopes()  # New in pymsalruntime 0.3.x
+    if granted_scopes:
+        return_value["scope"] = " ".join(granted_scopes)  # Mimic the on-the-wire data format
+    return return_value
+
+
+def _signin_silently(authority, client_id, scopes, **kwargs):
+    params = pymsalruntime.MSALRuntimeAuthParameters(client_id, authority)
+    params.set_requested_scopes(scopes)
+    callback_data = _CallbackData()
+    for k, v in kwargs.items():  # This can be used to support domain_hint, max_age, etc.
+        if v is not None:
+            params.set_additional_parameter(k, str(v))
+    pymsalruntime.signin_silently(
+        params,
+        "correlation", # TODO
+        lambda result, callback_data=callback_data: callback_data.complete(result))
+    callback_data.signal.wait()
+    return _convert_result(callback_data.auth_result, client_id)
+
+
+def _signin_interactively(
+        authority, client_id, scopes,
+        window=None,
+        prompt=None,
+        login_hint=None,
+        claims=None,
+        **kwargs):
+    params = pymsalruntime.MSALRuntimeAuthParameters(client_id, authority)
+    params.set_requested_scopes(scopes)
+    params.set_redirect_uri("placeholder")  # pymsalruntime 0.1 requires non-empty str,
+        # the actual redirect_uri will be a value hardcoded by the underlying WAM
+    if prompt:
+        if prompt == "select_account":
```

Comment:
Are you referring to the adjustment of the `params.set_select_account_option(pymsalruntime.SelectAccountOption.SHOWLOCALACCOUNTSCONTROL)` mechanism? Yes, MSAL Python would need to change this part, as soon as the underlying changes is shipped in a newer version of MsalRuntime.

Reply:
Yes, I meant pymsalruntime.SelectAccountOption.SHOWLOCALACCOUNTSCONTROL. 
Is that possible for you to remove this first so that when a new version of msalruntime is published, you don't have a build break?
Just to make sure since you may already send this topic branch to Azure CLI team for testing before MSALRuntime publish a new version.

Reply:
If we remove it here right now, it will mean one more feature gap in our partner teams' integration test.

But your point of the upcoming breaking change is legit. The right way to solve this will be for MSAL Python to pin the upper bound of MsalRuntime's version to `<0.6`. I'll make that change in my next commit.